### PR TITLE
Changing dependency from python3-distutils to python3-distutils-extra

### PR DESCRIPTION
**Pull request by davidjumani**
## Description
Changing dependency from python3-distutils to python3-distutils-extra to continue supporting Ubuntu 16.04

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
Kicking tests manually

**Comment by rhtyd** [comment]
Trillian fails to deploy env again u18 env:
```
20:24:59 TASK [kvm : Updated /etc/default/libvirt-bin in trusty] ************************
20:24:59 skipping: [pr4305-t2665-kvm-ubuntu-kvm1] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}
20:24:59 skipping: [pr4305-t2665-kvm-ubuntu-kvm2] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}
20:24:59 
20:24:59 TASK [kvm : Update /etc/default/libvirtd in bionic] ****************************
20:24:59 fatal: [pr4305-t2665-kvm-ubuntu-kvm2]: FAILED! => {"changed": false, "failed": true, "msg": "Destination /etc/default/libvirtd does not exist !", "rc": 257}
20:25:00 
20:25:00 msg: Destination /etc/default/libvirtd does not exist !
20:25:00 fatal: [pr4305-t2665-kvm-ubuntu-kvm1]: FAILED! => {"changed": false, "failed": true, "msg": "Destination /etc/default/libvirtd does not exist !", "rc": 257}
20:25:00 
```

**Comment by davidjumani** [comment]
@rhtyd Looks like an unrelated issue. Will try kicking tests on 16 and then 18

File: debian/control [hunk 1]
```diff
@@ -9,7 +9,7 @@ Homepage: http://www.cloudstack.org/
 
 Package: cloudstack-common
 Architecture: all
-Depends: ${misc:Depends}, ${python3:Depends}, genisoimage, nfs-common, python3-pip, python3-distutils, python3-netaddr
+Depends: ${misc:Depends}, ${python3:Depends}, genisoimage, nfs-common, python3-pip, python3-distutils-extra, python3-netaddr
```
Comment (shwstppr) [inline debian/control:12]:
```suggestion
Depends: ${misc:Depends}, ${python3:Depends}, genisoimage, nfs-common, python3-pip, python3-distutils | python3-distutils-extra, python3-netaddr
```
@davidjumani will this affect other Ubuntu releases? Saw this other PR, not sure if adding pipe will work here

Comment (davidjumani) [reply debian/control:12]: @shwstppr Thanks. I can test with these changes. Am currently running tests on Ubuntu 18 to check if everything works fine!

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@davidjumani can you manually kick tests and see if that fixes the issue for Ubuntu 16.04 and passes on Ubuntu 18.04+ as well

**Comment by davidjumani** [comment]
@rhtyd The envs are sporadically failing while running tests. Kicking them again

**Comment by davidjumani** [comment]
Kicking tests again

**Comment by shwstppr** [comment]
@davidjumani will it be better to merge latest master first? There have been some fixes for ubuntu on master lately

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@davidjumani ping, update?

**Comment by davidjumani** [comment]
@rhtyd Works fine on u20 and u18, snapshots are failing on u16. Looking into it

**Comment by rhtyd** [comment]
Ignore snapshot failures on Ubuntu 16.04 which could be due to old libvirt/qemu-img binary. Other than that is the PR working and fixes the blocker?

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
I think snapshot failure is a known issue on older distros with older libvirt and qemu-img; earlier snapshots were disabled by default if I recall correctly, there is a global setting to enable/disable snapshot feature on KVM.

**Comment by rhtyd** [comment]
@davidjumani which test results are for Ubuntu 16.04? and which are for 18.04 and 20.04?

**Comment by davidjumani** [comment]
> Trillian test result (tid-2768)
> Environment: kvm-ubuntu (x2), Advanced Networking with Mgmt server u
> Total time taken: 57788 seconds
> Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4305-t2768-kvm-ubuntu.zip
> Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_supported_versions.py
> Intermittent failure detected: /marvin/tests/smoke/test_list_ids_parameter.py
> Intermittent failure detected: /marvin/tests/smoke/test_pvlan.py
> Intermittent failure detected: /marvin/tests/smoke/test_resource_accounting.py
> Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
> Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
> Intermittent failure detected: /marvin/tests/smoke/test_usage.py
> Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
> Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
> Smoke tests completed. 78 look OK, 7 have error(s)
> Only failed tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_01_add_delete_kubernetes_supported_version 	`Error` 	1805.60 	test_kubernetes_supported_versions.py
> ContextSuite context=TestListIdsParams>:setup 	`Error` 	0.00 	test_list_ids_parameter.py
> test_create_pvlan_network 	`Error` 	0.05 	test_pvlan.py
> test_01_snapshot_root_disk 	`Error` 	10.21 	test_snapshots.py
> test_02_list_snapshots_with_removed_data_store 	`Error` 	29.31 	test_snapshots.py
> test_01_snapshot_usage 	`Error` 	7.37 	test_usage.py
> test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL 	`Failure` 	461.93 	test_vpc_redundant.py
> test_04_rvpc_network_garbage_collector_nics 	`Error` 	3897.41 	test_vpc_redundant.py
> test_hostha_kvm_host_fencing 	`Error` 	176.28 	test_hostha_kvm.py

@rhtyd u16



> Trillian test result (tid-2748)
> Environment: kvm-ubuntu18 (x2), Advanced Networking with Mgmt server u18
> Total time taken: 65497 seconds
> Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4305-t2748-kvm-ubuntu18.zip
> Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_supported_versions.py
> Intermittent failure detected: /marvin/tests/smoke/test_templates.py
> Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
> Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
> Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
> Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
> Smoke tests completed. 81 look OK, 4 have error(s)
> Only failed tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_01_add_delete_kubernetes_supported_version 	`Error` 	1804.59 	test_kubernetes_supported_versions.py
> test_11_migrate_vm 	`Error` 	49.88 	test_vm_life_cycle.py
> test_14_secure_to_secure_vm_migration 	`Error` 	129.61 	test_vm_life_cycle.py
> test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL 	`Failure` 	479.68 	test_vpc_redundant.py
> test_04_rvpc_network_garbage_collector_nics 	`Error` 	3954.98 	test_vpc_redundant.py
> test_05_rvpc_multi_tiers 	`Failure` 	443.88 	test_vpc_redundant.py
> test_disable_oobm_ha_state_ineligible 	`Error` 	1514.27 	test_hostha_kvm.py

@rhtyd u18
U20 tested locally since lab is having issues bringing up the env

**Comment by rhtyd** [comment]
tests LGTM

**Review by Pearl1594 — APPROVED** [review]
LGTM

**Review by shwstppr — APPROVED** [review]
LGTM based on code changes and installation check on different Ubuntu releases

**Pull request merged**